examples/reference/custom_components/ReactiveESM.md

```diff
@@ -0,0 +1,283 @@
+# `ReactiveESM`
+
+`ReactiveESM` makes it easy to create custom Panel components using JavaScript.
+
+```python
+import panel as pn
+import param
+
+pn.extension()
+
+class CounterButton(pn.ReactiveESM):
+
+    value = param.Integer()
+
+    _esm = """
+    export function render({ data }) {
+        let btn = document.createElement("button");
+        btn.innerHTML = `count is ${data.value}`;
+        btn.addEventListener("click", () => {
+            data.value+=1
+        });
+        data.watch(() => {
+            btn.innerHTML = `count is ${data.value}`;
+          }, 'value')
+        return btn
+    }
+    """
+
+CounterButton().servable()
+```
+
+:::{note}
+
+`ReactiveESM` was introduced June 2024 to be the successor of `ReactiveHTML`.
+
+`ReactiveESM` is very similar to [`AnyWidget`](https://anywidget.dev/), but `ReactiveESM` is optimized for usage with Panel.
+
+:::
+
+## Api
+
+### ReactiveESM Attributes
+
+* **`_esm`** (str | PurePath): An [ECMAScript module](https://nodejs.org/api/esm.html#modules-ecmascript-modules) string or a path pointing to a [ECMAScript module](https://nodejs.org/api/esm.html#modules-ecmascript-modules). If you are developing in a notebook or with `--autoreload` the file will automatically be reloaded when saved. The ECMAScript module should export a `render` function.
+* **`_import_map`** (dict): An [import map](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap) allows you to control how module specifiers are resolved.
+* **`_stylesheets`** (optional list of strings): An Optional  list of CSS strings.
+
+#### `render` Function
+
+The `render` function can accept the following arguments:
+
+- **`children`**: DON'T KNOW HOW TO EXPLAIN
+- **`data`**: The `data` element corresponds to the custom components Python Parameters and provides methods to `.watch` for parameter changes and `.send_event` back to Python.
+- **`el`**: The main html element of the component. Append elements to this element to display them. IS THIS OBSOLETE AS YOU SHOULD BE RETURNING THE ELEMENT TO RENDER?
```

Comment:
Not obsolete, there's a number of reasons you might want to insert additional nodes.